Updates to BL5340 DVK

ahedin25 opened the pull request.

jfischer-no commented on samples/subsys/display/lvgl/boards/bl5340_dvk_cpuapp.conf:
```diff
@@ -0,0 +1,3 @@
+CONFIG_KSCAN=y
+
+CONFIG_LV_Z_POINTER_KSCAN=y
```
Please justify why this overlay is needed and can not be resolved at the board level.
You commit message is wrong.

ahedin25 replied:
So I should add it to the Kconfig.defconfig?

ahedin25 committed:
boards: arm: bl5340_dvk: Update display configuration

Add display to default configuration.

Update board file to work with changes to graphics library.
Enable compatibility with new version of LVGL.
Prevent hard fault when running LVGL sample.
Add alias for touch controller.

Signed-off-by: Andrew Hedin <andrew.hedin@lairdconnect.com>

jfischer-no commented on boards/arm/bl5340_dvk/Kconfig.defconfig:
```diff
@@ -99,6 +99,53 @@ config HEAP_MEM_POOL_SIZE
 config BT_HCI_VS
 	default y if BT
 
+# BUYDISPLAY_2_8_TFT_TOUCH_ARDUINO
```
You nailed my Friday.

jfischer-no replied:
Maybe a clear description if any required at all, but not a label you just copied from the shield support.
If it is intended to be removable by the user, it should not be hardcoded into the board support in this way.

ahedin25 commented:
Getting a twister failure after rebase.

> zephyr/build/zephyr/misc/generated/extra_kconfig_options.conf:3: warning: malformed string literal in assignment to BT_DEVICE_NAME (defined at subsys/bluetooth/host/Kconfig:758). Assignment ignored.

nordicjm commented on boards/arm/bl5340_dvk/Kconfig.defconfig:
```diff
@@ -83,3 +80,87 @@ endif # BOARD_BL5340_DVK_CPUAPP || BOARD_BL5340_DVK_CPUAPP_NS
+config LV_COLOR_16_SWAP
+	default y
+
+config KSCAN
+	default y
+
+config INPUT
+	default y
+
+config LV_Z_POINTER_KSCAN
+	default y
```
input has replaced kscan - @fabiobaltieri for comment on this

fabiobaltieri replied:
It's in progress, LVGL still takes kscan events. Will write an lvgl pointer driver using the input APIs down the road but for now this is still the way to go.

More importantly:

```
# 2.8" TFT Touch Shield for Arduino w/Capacitive Touch Screen Module
```

If the display is on a shield this should not go in the board dts at all. Instead the board should define an `arduino-header-r3` node and then the display can be enabled with the generic shield code that already exists in `boards/shields/buydisplay_2_8_tft_touch_arduino`.

rerickson1 replied:
Our board has the display attached already and takes work for a user to remove it. See here:
https://www.lairdconnect.com/wireless-modules/bluetooth-modules/bluetooth-5-modules/bl5340-series-multi-core-bluetooth-52-802154-nfc-modules

fabiobaltieri replied:
Ok then this makes sense.

nordicjm commented on boards/arm/bl5340_dvk/bl5340_dvk_cpuapp_common.dts:
```diff
@@ -215,8 +219,8 @@
 		reset-gpios = <&gpio0 6 GPIO_ACTIVE_LOW>;
 		cmd-data-gpios = <&gpio0 12 GPIO_ACTIVE_LOW>;
 		rotation = <270>;
-		width = <240>;
-		height = <320>;
+		width = <320>;
+		height = <240>;
```
From https://www.buydisplay.com/2-8-inch-tft-touch-shield-for-arduino-w-capacitive-touch-screen-module: `ER-TFTM028-4 is 240x320 dots 2.8" color tft lcd module display with` - has this been tested and works?

ahedin25 replied:
The sample app hard faults without this change.  Maybe LV_Z_POINTER_KSCAN_SWAP_XY works differently than before?

fabiobaltieri replied:
probably kscan pointing at the ft5336 driver that does not implement an API anymore, see my other comment, you can add a `zephyr,kscan-input` to make it work again and point `kscan_input` to that instead

nordicjm replied:
Has this been tested and working since the values are flipped?

ahedin25 replied:
Yes. It has been tested with the lvgl sample app.

nordicjm commented on boards/arm/bl5340_dvk/bl5340_dvk_cpunet_common.dts:
```diff
@@ -0,0 +1,72 @@
+			reg = <0x00022000 0x16000>;
+		};
+		/* 12K */
+		scratch_partition: partition@38000 {
+			label = "image-scratch";
+			reg = <0x00038000 0x3000>;
+		};
```
Can go

ahedin25 committed:
boards: arm: bl5340_dvk: Prepare for addition of PA version

Add bt-hci-rpmsg-ipc to DTS.
Move common items to their own files.
Align debug setup with nrf5340 DK.
Remove scratch partition.

Signed-off-by: Andrew Hedin <andrew.hedin@lairdconnect.com>
